frontend/packages/ceph-storage-plugin/src/components/converged-credentials/credentials.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import { k8sGet } from '@console/internal/module/k8s';
+import { SecretModel, ConfigMapModel } from '@console/internal/models';
+import { createModalLauncher, ModalComponentProps } from '@console/internal/components/factory';
+import { OCS_NS } from '../../constants';
+
+const parseMonEndpoints = (data: string): string => {
+  return data.split(',')[0];
+};
+
+const createJsonFile = (data: Credentials) =>
+  `data:text/json;charset=utf-8,${encodeURIComponent(JSON.stringify(data))}`;
+
+const ExportCredentials: React.FC<ModalComponentProps> = (props) => {
+  const ref = React.useRef<HTMLAnchorElement>(null);
+  const [payload, setPayload] = React.useState('');
+  React.useEffect(() => {
+    const promises = [];
+    promises.push(k8sGet(SecretModel, 'rook-ceph-mon', OCS_NS));
+    promises.push(k8sGet(ConfigMapModel, 'rook-ceph-mon-endpoints', OCS_NS));
+    Promise.all(promises)
+      .then((data) => {
+        const fsid = data[0]?.data?.fsid;
+        const adminSecret = data[0]?.data['admin-secret'];
+        // const monSecret = data[0]?.data['mon-secret'];
+        const monIP = parseMonEndpoints(data[1]?.data?.data);
+        const namespace = OCS_NS;
+        setPayload(
+          createJsonFile({
+            admin: adminSecret,
+            monData: monIP,
+            ns: namespace,
+            fsid,
+          }),
+        );
+        ref.current.click();
+        props.close();
```

Comment:
So when both promises are resolved, component's state is updated (`setPayload`) which causes re-render, we programatically trigger the JSON file download and then we close the modal? Is this the expected behavior?

In other words, it seems that regardless of whether the promises succeed or not, the modal is always immediately closed.

Reply:
Yes. That is the exact workflow. Actually we do not want a modal. All we want is the file to download. But we are using a modal because it handles the injection of the component in the dom via `modal-component` easily.